edit-post/components/layout/index.js

```diff
@@ -112,26 +115,31 @@ function Layout( {
 				openedGeneralSidebar !== null && <GeneralSidebar
 					openedGeneralSidebar={ openedGeneralSidebar } />
 			}
+			<Popover.MobileScrollLock enabled={ layoutHasOpenSidebar && isMobileViewport } />
```

Comment:
There is also `ifViewportMatch` HOC which might be used to wrap this component. In that case, it would never mount when the viewport is larger than small (mobile). It leaves `layoutHasOpenSidebar` part in here. My question is if we can convert this component into HOC that would be applied to the popover instead? In that case we would naturally combine two related concepts. This `MobileScrollLock` would be a sibling component that renders only on mobile and when any popover is actually rendered.

Reply:
I would love to use `ifViewportMatch` here, but my understanding is that it depends on resolving a circular dependency between `@wordpress/components` and `@wordpress/viewport` (see #5316). We can probably get most of the way there though:

1. I can update `MobileScrollLock` to be enabled if present.
2. Make users of `MobileScrollLock` responsible for only adding for mobile viewports. I believe we only use it in two places right now.

Once we can use `ifViewportMatch`, we can push that requirement into `MobileScrollLock`.